manager-bundle/src/Dotenv/DotenvDumper.php

```diff
@@ -54,25 +58,53 @@ public function setParameters(array $params): void
 
     public function unsetParameter(string $name): void
     {
-        unset($this->parameters[$name]);
+        unset($this->setParameters[$name]);
+        $this->unsetParameters[] = $name;
     }
 
     public function dump(): void
     {
-        // Remove the .env file if there are no parameters
-        if (0 === \count($this->parameters)) {
-            $this->filesystem->remove($this->dotenvFile);
+        $file = '';
+        $lines = [];
 
-            return;
+        if (file_exists($this->dotenvFile)) {
+            $lines = preg_split('/\r\n|\r|\n/', file_get_contents($this->dotenvFile));
```

Comment:
Any reason for not using `file()`?

Reply:
Doesn’t support `\r` AFAIK.
But Symfony seems to support it in https://github.com/symfony/symfony/blob/6222f8eaf8498f75fcc018157386d7290a68d0bd/src/Symfony/Component/Dotenv/Dotenv.php#L230